javascript/MLE-Demo/book-schema.mjs

```diff
@@ -13,151 +13,88 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import {
-    GraphQLSchema,
-    GraphQLObjectType,
-    GraphQLString,
-    GraphQLInt,
-    GraphQLFloat,
-    GraphQLList
-  } from 'graphql';
-  import 'mle-js-fetch'; 
-  import oracledb from 'mle-js-oracledb';
-  
-  const BASE_URL = "https://www.googleapis.com/books/v1/volumes?q=";
-  
-  const getBooks = async (SEARCH_URL) => {
+import { makeExecutableSchema } from 'graphql-tools-schema';
+import 'mle-js-fetch';
+import oracledb from 'mle-js-oracledb';
+
+const BASE_URL = "https://www.googleapis.com/books/v1/volumes?q=";
+
+const typeDefs = `
+  type ImageLinks {
+    thumbnail: String
+  }
+
+  type Author {
+    name: String
+  }
+
+  type VolumeInfo {
+    title: String
+    subtitle: String
+    publisher: String
+    description: String
+    authors: [Author]
+    publishedDate: String
+    imageLinks: ImageLinks
+    averageRating: Float
+    pageCount: Int
+    language: String
+  }
+
+  type Book {
+    id: String
+    kind: String
+    etag: String
+    volumeInfo: VolumeInfo
+    stock: Int
+  }
+
+  type Volumes {
+    kind: String
+    totalItems: Int
+    items: [Book]
+  }
+
+  type Query {
+    library(research: String): Volumes
+  }
+`;
+
+const resolvers = {
+  Query: {
+    library: async (_, { research }) => {
+      return await getBooks(research);
+    }
+  },
+  VolumeInfo: {
+    authors: (volumeInfo) => volumeInfo.authors || [],
+  },
+  Book: {
+    stock: async (book) => {
+      const conn = oracledb.defaultConnection();
+      const result = await conn.execute('SELECT stock FROM inventory WHERE id = :id', [book.id], { outFormat: oracledb.OUT_FORMAT_OBJECT });
+      if (result.rows.length > 0) {
+        return result.rows[0].STOCK;
+      } else {
+        return 0;
+      }
+    }
+  },
+};
+
+const getBooks = async (SEARCH_URL) => {
   const conn = oracledb.defaultConnection();
-  conn.execute(`
-  begin
-     utl_http.set_wallet('file:/<absolute-path-to-your-wallet-directory>');
-  end;
+  await conn.execute(`
+    begin
+      utl_http.set_wallet('file:/<absolute-path-to-your-wallet-directory>');
+    end;
   `);
-  
   const response = await fetch(`${BASE_URL}${SEARCH_URL}`, { credentials: "include" });
```

Comment:
beginning with Oracle Database 23.3 it should no longer be necessary to use `createntials: include`